src/main/java/sg/ncl/MainController.java

```diff
@@ -2514,6 +2516,40 @@ public String getTopology(@PathVariable String teamName, @PathVariable String ex
         }
     }
 
+    @RequestMapping(value = "/request_internet/{teamName}/{teamId}/{expId}", method = RequestMethod.POST)
+    public String internetRequest(@PathVariable String teamName,
+                                  @PathVariable String teamId,
+                                  @PathVariable String expId,
+                                  @ModelAttribute InternetRequestForm internetRequestForm,
+                                  final RedirectAttributes redirectAttributes
+                                 ) throws WebServiceRuntimeException {
+
+        Realization realization = invokeAndExtractRealization(teamName, Long.parseLong(expId));
+        if(!realization.getState().equals(RealizationState.RUNNING.toString())) {
+            log.warn("Trying to request internet for the experiment {} from team {} with state {}", expId, teamName,realization.getState());
+            redirectAttributes.addFlashAttribute(MESSAGE, "Experiment " + realization.getExperimentName() + " need to be started before you can request for internet access" );
+            return "redirect:/experiments";
+        }
+
+        log.info("Requesting internet access at " + properties.requestInternetExperiment(teamId, expId));
+        JSONObject requestObject = new JSONObject();
+        requestObject.put("reason", internetRequestForm.getReason());
+        try {
+            HttpEntity<String> request = createHttpEntityWithBody(requestObject.toString());
+            restTemplate.setErrorHandler(new MyResponseErrorHandler());
+            ResponseEntity response = restTemplate.exchange(properties.requestInternetExperiment(teamId, expId),
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "response". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "response" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)